packages/astro/src/core/render/context.ts

```diff
@@ -51,12 +55,29 @@ export async function createRenderContext(
 		logging: options.env.logging,
 		ssr: options.env.ssr,
 	});
-	return {
+
+	let context = {
 		...options,
 		origin,
 		pathname,
 		url,
 		params,
 		props,
 	};
+
+	// We define a custom property, so we can check the value passed to locals
+	Object.defineProperty(context, 'locals', {
```

Comment:
can you add `enumerable: true`? Sometimes get support questions where people try to console.log the context.